contracts/implementation/Resolver.sol

```diff
@@ -0,0 +1,37 @@
+//SPDX-License-Identifier: CC-BY-NC-ND-4.0
+pragma solidity 0.8.7;
+
+import "../interfaces/IResolver.sol";
+import "../interfaces/IPoolFactory.sol";
+import "../interfaces/IPoolKeeper.sol";
+
+/// @title The resolver contract is responsible for periodically checking if a pool requires to be upkept
+/// @notice This contract will be used in gelato app to create a gelato task for upkeeping pools.
+contract Resolver is IResolver {
+    IPoolFactory public poolFactory;
+    IPoolKeeper public poolKeeper;
+
+    constructor(address _poolFactory) {
+        poolFactory = IPoolFactory(_poolFactory);
+        poolKeeper = IPoolKeeper(poolFactory.getPoolKeeper());
+    }
+
+    /**
+     * @notice Iterate through all pools and will then return true if can be upkept, and data including the function selector and parameters.
+     * @return execPayLoad payload data of the `isUpkeepRequiredSinglePool` function selector with the pool address as argument
+     */
+    function upKeepChecker() external view override returns (bytes[] memory execPayLoad) {
```

Comment:
Based off this: https://docs.gelato.network/guides/writing-a-resolver/smart-contract-resolver
I think `IResolver` must implement a function with signature `function checker() external view override returns (bool canExec, bytes memory execPayload)`